core/server/api_container/server/service_network/default_service_network.go

```
@@ -1761,3 +1780,23 @@ func renderTemplateToFile(templateAsAString string, templateData interface{}, de
 	}
 	return nil
 }
+
+// generateUniqueNameForFileArtifact - this method returns unique file artifact name after x max retries
+func generateUniqueNameForFileArtifact(
+	checkIfFileArtifactNameExists func(artifactName string) bool,
+	generateNatureThemeName func() string,
+	maxRetry int) (string, error) {
+
+	maybeUniqueName := generateNatureThemeName()
+
+	for !checkIfFileArtifactNameExists(maybeUniqueName) && maxRetry > 0 {
+		maybeUniqueName = generateNatureThemeName()
+		maxRetry = maxRetry - 1
+	}
+
+	if maxRetry == 0 {
```

I don't think we can fail hard here if the max retry is hit. Having a nice and clear name for artifact it seems like a "nice to have" feature that is not worth failing the entire `kurtosis run` just because we were unable to generate a name.
I think we should pick a maxRetry appropriate to our goals here. IDK how the random name generator works, but let's target for example 100K artifacts in total, and pick a maxRetry such that it won't be hit in 99% of the cases. Something like this.
And then when it's hit, let's log a big warning, but fallback to a random UUID, or something we _know_ will work, to avoid failing the entire run.